### PR TITLE
[BugFix] Limit bytes returned by left_bytes_string

### PR DESCRIPTION
**Pull request by rickif**
## What I'm doing:
The `JsonParser::left_bytes_string(size_t sz)` returns all left bytes now, while the argument `sz` dose not work at all.

Fixes 

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/exec/json_parser.cpp [hunk 1]
```diff
@@ -90,7 +90,9 @@ std::string JsonDocumentStreamParser::left_bytes_string(size_t sz) noexcept {
     if (off >= _len) {
         return {};
     }
-    return std::string(reinterpret_cast<char*>(_data) + off, _len - off);
+
+    auto len = std::min(_len - off, sz);
+    return std::string(reinterpret_cast<char*>(_data) + off, len);
```
Comment (jaogoy) [inline be/src/exec/json_parser.cpp:95]: why not extract a left_bytes_string() for both parsers?

less code, more stable.

Comment (rickif) [reply be/src/exec/json_parser.cpp:95]: `JsonArrayParser` and `JsonDocumentStreamParser` share the interface `JsonParser:left_bytes_string`.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by rickif** [comment]
@mergifyio rebase

**Review by meegoo — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by srlch** [comment]
@Mergifyio backport branch-3.3

**Comment by srlch** [comment]
@Mergifyio backport branch-3.2

**Comment by srlch** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**